helper.go

```diff
@@ -0,0 +1,53 @@
+package pubsub
+
+import (
+	"time"
+
+	amqp "github.com/rabbitmq/amqp091-go"
+)
+
+func SetMessageHeader(msg *amqp.Delivery, key string, value interface{}) {
+	if msg == nil {
+		return
+	}
+
+	if msg.Headers == nil {
+		msg.Headers = make(amqp.Table)
+	}
+
+	msg.Headers[key] = value
+}
+
+func GetMessageHeader[T any](msg *amqp.Delivery, key string) (value T, ok bool) {
+	if msg == nil {
+		return
+	}
+
+	if msg.Headers == nil {
+		return
+	}
+
+	value, ok = msg.Headers[key].(T)
+
+	return value, ok
+}
+
+func GetNumberOfAttempts(msg *amqp.Delivery) (int, bool) {
+	return GetMessageHeader[int](msg, "x-retry-attempts")
```

Comment:
Call [getAttemptsFromHeaders](https://github.com/samber/go-amqp-pubsub/blob/main/retry_acknowledgement.go#L18) instead.

For better serialization, we convert `x-retry-attempts` value to a string.